lib/Demangling/NodePrinter.cpp

```diff
@@ -3159,6 +3160,75 @@ void NodePrinter::printEntityType(NodePointer Entity, NodePointer type,
+                                          size_t MangledNameLength) {
+    std::string invalid = "";
+    std::string unlabelledArg = "_: ";
+    Context ctx;
+    NodePointer root = ctx
+        .demangleSymbolAsNode(StringRef(MangledName, MangledNameLength));
+    if (!root)
+        return invalid;
+    if (root->getNumChildren() >= 1) {
+        NodePointer firstChild = root->getChild(0);
+        if (firstChild->getKind() == Node::Kind::KeyPathGetterThunkHelper) {
+            NodePointer child = firstChild->getChild(0);
+            if (child->getKind() == Node::Kind::Subscript) {
+                std::string subscriptText = "subscript(";
+                child = child->getChild(1);
+                if (child != nullptr) {
+                    // There is an argument label:
+                    if (child->getKind() == Node::Kind::LabelList) {
+                        size_t idx = 0;
+                        size_t numChildren = child->getNumChildren();
+                        if (numChildren == 0) {
+                            subscriptText += unlabelledArg;
+                        } else {
+                            while (idx < numChildren) {
+                                Node *argChild = child->getChild(idx);
+                                idx += 1;
+                                if (argChild->getKind() == Node::Kind::Identifier) {
+                                    subscriptText += std::string(argChild->getText()) + ": ";
+                                } else if (argChild->getKind() == Node::Kind::FirstElementMarker ||
+                                           argChild->getKind() == Node::Kind::VariadicMarker) {
+                                    subscriptText += unlabelledArg;
+                                }
+                            }
+                        }
+                    }
+                } else {
+                    subscriptText += unlabelledArg;
+                }
+                return subscriptText + ")";
+            } else if (child->getKind() == Node::Kind::Variable) {
+                child = child->getChild(1);
+                if (child == nullptr) {
+                    return invalid;
+                }
+                if (child->getKind() == Node::Kind::PrivateDeclName) {
+                    child = child->getChild(1);
+                    if (child == nullptr) {
+                        return invalid;
+                    }
+                    if (child->getKind() == Node::Kind::Identifier) {
+                        return std::string(child->getText());
+                    }
+                } else if (child->getKind() == Node::Kind::Identifier){
+                    return std::string(child->getText());
+                }
+                return invalid;
+            } else {
+                return invalid;
+            }
+        } else {
+            return invalid;
+        }
+    } else {
+        return invalid;
+    }
+    return invalid;
```

Comment:
I think instead of having all these else statements, you can simply have the outermost one and remove the inner ones because they'll all escape outside.